sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,94 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=5, normalize=True):
```

Comment:
Thinking about the default, should it be `2`, e.g. the minimum value for which the function can be called? It would work in all cases. 5 would fail if n_classes < 5

Reply:
Agree with that!

Reply:
I am not sure that we want `ValueError` [here](https://github.com/scikit-learn/scikit-learn/pull/16625/files#diff-2e8e8eee8d9982a7796bdad9558a3482R1502). I believe that for `k>=n_classes` we should raise a warning with the same message probably but we probably want to return a score. No?

Reply:
I think we should raise an error if k >= n_classes. This will always output an accuracy of 1 which will lure inexperienced users into thinking their estimator is doing great, where in reality they're just misusing a metric.

In general, we try to make misuses hard / impossible

Reply:
> In general, we try to make misuses hard / impossible  

Agree. 

> will lure inexperienced users into thinking their estimator is doing great

If  they have a binary problem and they ask for top_5 accuracy then their estimator will be doing great obviously!

I dunno. IMHO a warning is enough.

Reply:
>   If they have a binary problem and they ask for top_5 accuracy then their estimator will be doing great obviously!

Indeed. Also +1 to increase the default to at least 3. I think top_5 is good when you have a very large number of classes (e.g. ImageNet), but 30-60 classes which is more common with tabular data, top_3 is already quite useful.

>

Reply:
> If they have a binary problem and they ask for top_5 accuracy then their estimator will be doing great obviously!

Obvious to you, maybe not to them. Also note that we error in the binary case too.